src/doc/unstable-book/src/compiler-flags/crate-attr.md

```diff
@@ -0,0 +1,10 @@
+# `crate-attr`
+
+The tracking issue for this feature is: [#138287](https://github.com/rust-lang/rust/issues/138287).
+
+------------------------
+
+The `-Z crate-attr` flag allows you to inject attributes into the crate root.
+For example, `-Z crate-attr=crate_name="test"` acts as if `#![crate_name="test"]` were present before the first source line of the crate root.
```

Comment:
They are not inserted at the source level though, crate is parsed first, and immediately after that the `-Z crate-attr`s are inserted at the top (before any kind of expansion, including pre-configuration).
So they cannot mess with something like shebang or comments on the first line.

Reply:
i added a section on how this formally behaves. let me know if that matches what you'd expect.